src/payments/sisp.js

```diff
@@ -0,0 +1,98 @@
+require('dotenv').config();
+const { format: formatDate } = require('date-fns');
+const sha512 = require('js-sha512');
+const btoa = require('btoa');
+
+
+const CVE_CURRENCY_CODE = '132';
+
+const toBase64 = (u8) => btoa(String.fromCharCode.apply(null, u8));
+
+const sha512ToBase64 = (input) => toBase64(sha512.digest(input));
+
+const generateFingerprint = (
+  posAutCode,
+  timestamp,
+  amount,
+  merchantRef,
+  merchantSession,
+  posID,
+  currency,
+  transactionCode,
+  entityCode,
+  referenceNumber
+) => {
+  let toHash = `${sha512ToBase64(posAutCode) + timestamp + (Number(parseFloat(amount) * 1000))}${merchantRef}${merchantSession.trim()}${posID.trim()}${currency.trim()}${transactionCode.trim()}`;
+
+  if (entityCode) {
+    toHash += Number(entityCode.trim());
+  }
+  if (referenceNumber) {
+    toHash += Number(referenceNumber.trim());
+  }
+
+  return sha512ToBase64(toHash);
+};
+
+module.exports = {
+  /**
+ * GENERATE PAYMENT REQUEST FORM
+ * @param {String} referenceId - Required - This is the payment reference. In this case, we need to know which payment was processed when the SISP returns the payment response.
+ * @param {Number} total - Required - This is the amount SISP should process the payment.
+ * @returns {Document} response - HTML Form to process payments
+ */
+  generatePaymentForm: (referenceId, total) => {
+    const posID = process.env.SISP_POS_ID;
+    const posAutCode = process.env.SISP_POS_AUT_CODE;
+
+    const merchantSession = `S${formatDate(new Date(), 'yyyyMMddHHmmss')}`;
+    const dateTime = formatDate(new Date(), 'yyyy-MM-dd HH:mm:ss');
+
+    const responseUrl = process.env.SISP_WEBHOOK_URL;
+    const SISP_URL = 'https://mc.vinti4net.cv/Client_VbV_v2/biz_vbv_clientdata.jsp';
```

Comment:
I think that variable should be on the .env file

Reply:
@FabyoSK  This is not necessary because this URL is public from SISP documentation

Reply:
but we'll need to change the lib if the URL changes, it's less responsibility if we allow the dev to put the link by himself

Reply:
@FabyoSK  It makes sense, but if the URL changes, how the user will know because in this case they were using our lib, I think this is our responsibility to update the lib and make sure other things was not break by this and others changes.

Reply:
Okay, for me its okay